src/aosm/azext_aosm/tests/latest/unit_test/test_processors/test_core_arm_processor.py

```diff
@@ -0,0 +1,175 @@
+from unittest.mock import MagicMock
+from azext_aosm.build_processors.arm_processor import AzureCoreArmBuildProcessor
+from azext_aosm.inputs.arm_template_input import ArmTemplateInput
+from azext_aosm.vendored_sdks.models import (
+    AzureCoreNetworkFunctionArmTemplateApplication,
+    ApplicationEnablement, ArmResourceDefinitionResourceElementTemplate,
+    ArmResourceDefinitionResourceElementTemplateDetails,
+    ArmTemplateArtifactProfile,
+    ArmTemplateMappingRuleProfile,
+    AzureCoreArmTemplateArtifactProfile,
+    NSDArtifactProfile,
+    TemplateType,
+    AzureCoreArmTemplateDeployMappingRuleProfile,
+    DependsOnProfile,
+    ManifestArtifactFormat,
+    ReferencedResource,
+)
+from azext_aosm.common.constants import TEMPLATE_PARAMETERS_FILENAME
+from azext_aosm.definition_folder.builder.local_file_builder import LocalFileBuilder
+from azext_aosm.common.artifact import LocalFileACRArtifact
+
+code_directory = os.path.dirname(__file__)
+parent_directory = os.path.abspath(os.path.join(code_directory, "../.."))
+mock_vnf_directory = os.path.join(parent_directory, "mock_core_vnf")
+
+
+class AzureCoreArmProcessorTest(TestCase):
+    """Class to test AzureCore ARM Processor functionality"""
+
+    def setUp(self):
+        logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+        mock_arm_template_path = os.path.join(mock_vnf_directory, "ubuntu-template.json")
+        self.core_arm_input = ArmTemplateInput(
+            artifact_name="test-artifact-name",
+            artifact_version="1.1.1",
+            template_path=mock_arm_template_path,
+            default_config=None
+        )
+        self.processor = AzureCoreArmBuildProcessor("test-name", self.core_arm_input)
+
+    def test_get_artifact_manifest_list(self):
+        """Test get artifact manifest list for Azure Core arm processor."""
+        manifest_list = self.processor.get_artifact_manifest_list()
+        mock_manifest_artifact_format = ManifestArtifactFormat(
+            artifact_name="test-artifact-name",
+            artifact_type="ArmTemplate",
+            artifact_version="1.1.1",
+        )
+        self.assertEqual(len(manifest_list), 1)
+        self.assertIsInstance(manifest_list[0], ManifestArtifactFormat)
+        self.assertEqual(manifest_list[0], mock_manifest_artifact_format)
+
+    def test_artifact_details(self):
+        """Test get artifact details for Azure Core arm processor."""
+        artifact_details = self.processor.get_artifact_details()
+        mock_arm_template_path = os.path.join(mock_vnf_directory, "ubuntu-template.json")
+        mock_artifact = [LocalFileACRArtifact(
+            artifact_name="test-artifact-name",
+            artifact_type="ArmTemplate",
+            artifact_version="1.1.1",
+            file_path=mock_arm_template_path,
+        )]
+
+        # Testing each individial part of artifact are equal
+        self.assertEqual(artifact_details[0][0].artifact_name, mock_artifact[0].artifact_name)
```

Comment:
this makes me think a class for the return type of get_artifact_details might be useful. You'd might even reduce the amount of code by not needing to document the current Tuplie(list,list) in each get_artifact_details implementation, and it would make these assertions easier to understand